### PR TITLE
PIM-4596: Fix return value for orm repo

### PR DESCRIPTION
**Pull request by solivier**
| Q                    | A
| -------------------- | ---
| Bug fix?             |
| New feature?         |
| BC breaks?           |
| CI currently passes? | y
| Tests pass?          | y
| Scenarios pass?      | running
| Checkstyle issues?*  | n
| PMD issues?**        | n
| Changelog updated?   | n
| Fixed tickets        |
| DB schema updated?   |
| Migration script?    |
| Doc PR               |

http://core-ci.akeneo.com/job/pim-pr-behat/2941/console
http://core-ci.akeneo.com/job/pim-pr-behat-mongodb/1001/console

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Command/UpdateProductCommand.php [hunk 1]
```diff
@@ -313,8 +315,8 @@ protected function save(ProductInterface $product)
      */
     protected function createToken(OutputInterface $output, $username)
     {
-        $userManager = $this->getContainer()->get('oro_user.manager');
-        $user = $userManager->findUserByUsername($username);
+        $userRepository = $this->getContainer()->get('pim_user.repository.user');
+        $user = $userRepository->findOneByIdentifier($username);
```
Comment (phaseinducer) [inline src/Pim/Bundle/CatalogBundle/Command/UpdateProductCommand.php]: Sometimes we use the provider, sometimes the repo, we should decide.

Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Command/UpdateProductCommand.php]: Using directly the repo is better imo

Comment (phaseinducer) [inline src/Pim/Bundle/CatalogBundle/Command/UpdateProductCommand.php]: The advantage of using the provider is that it abstracts the way you fetch the user and it handles errors more precisely (like throwing `UsernameNotFoundException`).

**Comment by jjanvier** [comment]
gtm

**Pull request merged**